### PR TITLE
Remove the class FIRMessagingRmq2PersistentStore 

### PR DESCRIPTION
**Pull request by chliangGoogle**
The file FIRMessagingRmqManager is essentially calling the same functions of FIRMessagingRmq2PersistentStore, that becomes an unnecessary class layer.
Merging FIRMessagingRmq2PersistentStore into FIRMessagingRmqManager and remove the file FIRMessagingRmq2PersistentStore.

This is a clean up before fixing the issue: b/142357385
We need to move database operations off main thread and by reducing the unnecessary class layer helps us adding async call in the next PR.

minor: remove removeRmqMessagesWithRmqId: method as it's only used in test.

### CODE REVIEW

File: Firebase/Messaging/FIRMessagingDefines.h [hunk 1]
```diff
@@ -45,3 +45,14 @@ _Pragma("clang diagnostic pop")
     return nil; \
   } while (0)
 #endif
+
+#if defined(DEBUG)
+#define FIRMesssaging_MUST_NOT_BE_MAIN_THREAD()                                                 \
```
Comment (morganchen12) [inline Firebase/Messaging/FIRMessagingDefines.h]: nit: `Messsaging` -> `Messaging`

Comment (morganchen12) [inline Firebase/Messaging/FIRMessagingDefines.h]: Looks like this macro may be unused?

File: Firebase/Messaging/FIRMessagingDefines.h [hunk 2]
```diff
@@ -45,3 +45,14 @@ _Pragma("clang diagnostic pop")
...
+  do {                                                                                \
+    NSAssert(![NSThread isMainThread], @"Must not be executing on the main thread."); \
```
Comment (morganchen12) [inline Firebase/Messaging/FIRMessagingDefines.h]: Though it's debug-only, this assert may trigger false-positives if the main GCD queue is suspended. See http://blog.benjamin-encz.de/post/main-queue-vs-main-thread/

File: Firebase/Messaging/FIRMessagingDefines.h (hunk 1 above)
Comment (chliangGoogle) [reply Firebase/Messaging/FIRMessagingDefines.h]: Yeah I plan to use it in the next PR when I'm moving the db operation off main thread. I can remove from this PR if you like.

File: Firebase/Messaging/FIRMessagingDefines.h (hunk 2 above)
Comment (chliangGoogle) [reply Firebase/Messaging/FIRMessagingDefines.h]: Removed it for now.

File: Firebase/Messaging/FIRMessagingDefines.h (hunk 1 above)
Comment (morganchen12) [reply Firebase/Messaging/FIRMessagingDefines.h]: I think it's ok to leave in, but the macro in the `#else` statement should have the same name or it'll cause unrecognized identifier errors.

**Pull request merged**